pkg/controller/backupinfrastructure/backup_infrastructure_control.go

```diff
@@ -0,0 +1,372 @@
+				backupInfrastructureLogger.Errorf("Could not update the BackupInfrastructure status after deletion successful: %+v", updateErr)
+				return true, updateErr
+			}
+			if err := c.removeFinalizer(op); err != nil {
+				return true, err
+			}
+			return false, nil
+		}
+	}
+
+	// When a BackupInfrastructure deletion timestamp is not set we need to create/reconcile the backup infrastructure.
+	if updateErr := c.updateBackupInfrastructureStatus(op, gardenv1beta1.ShootLastOperationStateProcessing, operationType, "Reconciliation of Backup Infrastructure state in progress.", 1, nil); updateErr != nil {
+		backupInfrastructureLogger.Errorf("Could not update the BackupInfrastructure status after reconciliation start: %+v", updateErr)
+		return true, updateErr
+	}
+	if reconcileErr := c.reconcileBackupInfrastructure(op); reconcileErr != nil {
+		if updateErr := c.updateBackupInfrastructureStatus(op, gardenv1beta1.ShootLastOperationStateError, operationType, reconcileErr.Description+" Operation will be retried.", 1, reconcileErr); updateErr != nil {
+			backupInfrastructureLogger.Errorf("Could not update the BackupInfrastructure status after reconciliation error: %+v", updateErr)
+			return true, updateErr
+		}
+		return true, errors.New(reconcileErr.Description)
+	}
+	if operationType == gardenv1beta1.ShootLastOperationTypeCreate {
+		c.recorder.Eventf(backupInfrastructure, corev1.EventTypeNormal, gardenv1beta1.EventReconciled, "[%s] Reconciled Backup Infrastructure state", operationID)
+	}
+	if updateErr := c.updateBackupInfrastructureStatus(op, gardenv1beta1.ShootLastOperationStateSucceeded, operationType, "Backup Infrastructure has been successfully reconciled.", 100, nil); updateErr != nil {
+		backupInfrastructureLogger.Errorf("Could not update the Shoot status after reconciliation success: %+v", updateErr)
+		return true, updateErr
+	}
+
+	return true, nil
+}
+
+// reconcileBackupInfrastructure reconciles a BackupInfrastructure state.
+func (c *defaultControl) reconcileBackupInfrastructure(o *operation.Operation) *gardenv1beta1.LastError {
+
+	// We create botanists (which will do the actual work).
+	botanist, err := botanistpkg.New(o)
+	if err != nil {
+		return formatError("Failed to create a Botanist", err)
+	}
+	seedCloudBotanist, err := cloudbotanistpkg.New(o, common.CloudPurposeSeed)
+	if err != nil {
+		return formatError("Failed to create a Seed CloudBotanist", err)
+	}
+
+	var (
+		defaultRetry = 30 * time.Second
+
+		f                     = flow.New("Backup Infrastructure creation").SetProgressReporter(o.ReportBackupInfrastructureProgress).SetLogger(o.Logger)
+		deployBackupNamespace = f.AddTask(botanist.DeployBackupNamespaceFromBackupInfrastructure, defaultRetry)
+		_                     = f.AddTask(seedCloudBotanist.DeployBackupInfrastructure, 0, deployBackupNamespace)
```

Comment:
Why is that done by the cloud botanist? There isn't anything cloud specific, is it?

Reply:
Sorry, forgot to change here. Function definition is under `botanist` pkg only. Will change it.

Reply:
Yes it is. It deploys terraform job with cloud specific configuration/charts.

Reply:
True.